core/field_variable.ts

```diff
@@ -123,9 +123,9 @@ export class FieldVariable extends FieldDropdown {
    * Configure the field based on the given map of options.
    * @param config A map of options to configure the field based on.
    */
-  protected override configure_(config: AnyDuringMigration) {
+  protected override configure_(config: Config) {
     super.configure_(config);
-    this.setTypes_(config['variableTypes'], config['defaultType']);
+    this.setTypes_(config.variableTypes, config.defaultType);
```

Comment:
Perhaps this is best suited for another PR, though if the input is trusted to match its type contract, `setTypes_` can be updated to remove some of its type checks.

One reason *not* to remove the runtime type checks would be if a JS project using Blockly improperly used the types, though I think it's reasonable to assume the type contract is properly fulfilled in a TS environment. Since this change will be a major version bump, it shouldn't accidentally break any projects.

Reply:
Going to leave this as a follow-up.